gobblin-data-management/src/main/java/gobblin/data/management/copy/CopySource.java

```diff
@@ -93,8 +101,11 @@
 
         Path targetRoot = getTargetRoot(state, datasetFinder, copyableDataset);
 
-        Collection<Partition<CopyableFile>> partitions =
-            copyableDataset.partitionFiles(copyableDataset.getCopyableFiles(targetFs, targetRoot));
+        CopyConfiguration copyConfiguration = new CopyConfiguration(targetRoot,
```

Comment:
It is not mandatory. If it is null it will default to preserve nothing (this is done in PreserveAttributes itself).